### ISSUE TITLE
Toolbar and tabs doesn't appear when you swipe the ViewPager

### ISSUE THREAD

**Issue by vezikon** [florent37/MaterialViewPager#51]
If the header are hidden in the current ViewPager item when you scroll it doesn't appear again. 
I have manage to make it work somehow

- In the `MaterialViewPagerAnimator` I added two methods:

*`registerViewPager` allows to me to track the ViewPager's page change events

```Java
public void registerViewPager(PagerSlidingTabStrip viewPager) {
        viewPager.setOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {

            }

            @Override
            public void onPageScrollStateChanged(int state) {
                onPagerStateChanged();
            }
        });
    }
```

*`onPagerStateChanged` allows to me check the header state through `lastYOffset` parameter

```Java
public void onPagerStateChanged() {
if (ENABLE_LOG)
            Log.d(TAG, "scrollDown");

        if (lastYOffset > mHeader.toolbarLayout.getHeight()) {

            animateEnterToolbarLayout(lastYOffset);


        } else if (lastYOffset <= mHeader.toolbarLayout.getHeight()) {
            if (headerAnimator != null) {
                ViewHelper.setTranslationY(mHeader.toolbarLayout, 0);
                followScrollToolbarIsVisible = true;
            } else {
                headerYOffset = Float.MAX_VALUE;
                followScrollToolbarLayout(lastYOffset);
            }
        }
}
```

- In `MaterialViewPagerHelper` I added method to register the ViewPager to the animator:

```Java
 /**
     * Register a PagerSlidingTabStrip to the current MaterialViewPagerAnimator
     *
     * @param activity         current context
     * @param pagerSlidingTabStrip     the scrollable
     */
    public static void registerViewPager(Activity activity, PagerSlidingTabStrip pagerSlidingTabStrip) {
        if (activity != null && hashMap.containsKey(activity)) {
            MaterialViewPagerAnimator animator = hashMap.get(activity);
            if (animator != null) {
                animator.registerViewPager(pagerSlidingTabStrip);
            }
        }
    }
```

- Finally register the ViewPager in your activity:

```Java
        MaterialViewPagerHelper.registerViewPager(this, mViewPager.getPagerTitleStrip());
```

**Comment by florent37** [comment]
@all you think it'll be usefull to display again  the tabs when swipping ? I didn't thought :smile: 
if I got many upvote, I'll include it :+1:

**Comment by florent37** [comment]
btw thanks @vezikon for your work :+1:

**Comment by vezikon** [comment]
Your welcome 😉

**Comment by florent37** [comment]
thanks I included it in the 1.0.5 with the attribute viewpager_displayToolbarWhenSwipe :+1:
